recipes/mpdecimal/2.5.x/conanfile.py

```diff
@@ -105,9 +105,10 @@ def generate(self):
             tc.generate()
 
             deps = AutotoolsDeps(self)
-            if is_apple_os(self) and self.settings.arch == "armv8":
-                deps.environment.append("LDFLAGS", ["-arch arm64"])
-                deps.environment.append("LDXXFLAGS", ["-arch arm64"])
+            if is_apple_os(self):
```

Comment:
I appreciate this was already in the recipe, but I don't fully understand why it was needed.

Conan already adds `-arch xxx` when Cross building, and it should not be needed when *not* cross building in the sense that the compiler will already target the "native" architecture.
Pretty sure this can be removed, unless there's a CI log to look at?

Reply:
https://github.com/conan-io/conan-center-index/issues/23994

Points to a log on May 14th

Build: https://c3i.jfrog.io/c3i/misc-v2/logs/pr/23989/3-macos-clang/mpdecimal/2.5.1//9f69305d358977042aa4b289c18f45fcf5827ece-build.txt
Test: https://c3i.jfrog.io/c3i/misc-v2/logs/pr/23989/3-macos-clang/mpdecimal/2.5.1//9f69305d358977042aa4b289c18f45fcf5827ece-test.txt

The build log actually shows the exact invocation:
```
g++ -pthread -I. -I../libmpdec -Wall -Wextra -std=c++11 -pedantic -DNDEBUG -stdlib=libc++ -std=c++17 -m64 -O3 -isysroot /Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -arch x86_64 -fPIC -c decimal.cc -o .objs/decimal.o
g++ -pthread -L../libmpdec -dynamiclib -install_name @rpath/libmpdec++.3.dylib -undefined dynamic_lookup -compatibility_version 2.5 -current_version 2.5.1 -o libmpdec++.2.5.1.dylib .objs/decimal.o -lmpdec -lm
ld: warning: ignoring file ../libmpdec/libmpdec.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file .objs/decimal.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
```

Reply:
So maybe fully removing the section would fix it?

Reply:
Actually, I might know why. I don't think `LDXXFLAGS` is a "standard" variable (correct me if I'm wrong). So maybe instead of this, if `LDFLAGS` is properly set, we could add `LDFLAGS` as well as `LDXXFLAGS` when linking.

Reply:
I'll do some experiments locally and see.

Reply:
I can also fix it by doing the following:
```
if cross_building(self):
    deps.environment.append("LDXXFLAGS", ["$LDFLAGS"])
```
That might be a more general cross-building fix too.

Reply:
Yes! it's exactly that - Conan is setting up the right variables, but they're not used by the build script - that's why it only affects the C++ library but the C one is fine.

```
            tc = AutotoolsToolchain(self)
            tc.configure_args.append("--enable-cxx" if self.options.cxx else "--disable-cxx")
            tc_env = tc.environment()
            tc_env.append("LDXXFLAGS", ["$LDFLAGS"])
            tc.generate(tc_env)
```

Something like the above will work - no need to make it conditional, as Conan also sets up other variables that may be useful (e.g the sysroot when needed).

Reply:
Alternatively, I can add this in the build step (or add the equivalent as a patch)
```
replace_in_file(self, os.path.join(self.build_folder, "libmpdec++", "Makefile"), "$(LDXXFLAGS)", "$(LDXXFLAGS) $(LDFLAGS)")
```

@jcar87 thoughts on possible alternatives?

Reply:
Never mind, just saw your comment. I like that version, I'll update it in a minute.